synapse_data_warehouse/synapse_event/dynamic_tables/V2.53.0__node_event_table.sql

```diff
@@ -0,0 +1,65 @@
+USE SCHEMA {{ database_name }}.SYNAPSE_EVENT;
+
+CREATE OR REPLACE DYNAMIC TABLE node_event
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+        WITH unique_events AS (
+            SELECT
+                ID,
+                VERSION_NUMBER,
+                CHANGE_TYPE,
+                CHANGE_TIMESTAMP,
+                CHANGE_USER_ID,
+                NODE_TYPE,
+                PARENT_ID,
+                BENEFACTOR_ID,
+                PROJECT_ID,
+                CREATED_ON,
+                CREATED_BY,
+                MODIFIED_ON,
+                MODIFIED_BY,
+                FILE_HANDLE_ID,
+                NAME,
+                IS_PUBLIC,
+                IS_CONTROLLED,
+                IS_RESTRICTED,
+                EFFECTIVE_ARS,
+                ANNOTATIONS,
+                INTERNAL_ANNOTATIONS,
+                DERIVED_ANNOTATIONS,
+                VERSION_COMMENT,
+                VERSION_LABEL,
+                VERSION_HISTORY,
+                ACTIVITY_ID,
+                COLUMN_MODEL_IDS,
+                SCOPE_IDS,
+                ITEMS,
+                REFERENCE,
+                IS_SEARCH_ENABLED,
+                DEFINING_SQL,
+                ALIAS,
+                PROJECT_STORAGE_USAGE,
+                SNAPSHOT_DATE,
+                SNAPSHOT_TIMESTAMP
+            FROM
+                {{ database_name }}.synapse_raw.nodesnapshots --noqa: TMP
+            WHERE
+                -- only `id`, `change_*`, and `snapshot_*` columns are defined regardless of `change_type`
+                id is not null
+                and change_type is not null
+            QUALIFY ROW_NUMBER() OVER (
+                    -- `id` and `version_number` provide at least one record for each Synapse entity version.
+                    -- `change_type` ensures that simultaneous events are retained, regardless of their `change_type`.
+                    -- `modified_on` enables us to capture updates to Synapse Table entities and other
+                    -- update scenarios where the `version_number` does not change (like when we update annotations
+                    -- without incrementing the version). 
+                    PARTITION BY id, version_number, change_type, modified_on
```

Comment:
Should we use `etag` within the `PARTITION BY` statement rather than `modified_on`? My thought is based on the wording below that each etag update indicates that "Something" changed:

> Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle concurrent updates. Since the E-Tag changes every time an entity is updated it is used to detect when a client's current representation of an entity is out-of-date.

Reply:
@BryanFauble

I think they're equivalent: https://sagebionetworks.slack.com/archives/C05PBT2FUHF/p1747426131028369?thread_ts=1747334513.248759&cid=C05PBT2FUHF